Lesson 4 Assignments

bcamp3 opened the pull request:
- mailroom_pt2.py : Updated mail room cmd line script using dictionary as database
- trigrams.py : Trigrams script tested with short phrase, sherlock_small.txt, and sherlock.txt examples

charlesrobison commented on students/bcamp3/lesson04/mailroom_pt2.py:
```diff
@@ -0,0 +1,133 @@
+
+if __name__ == "__main__":
+    """Execute the main menu prompt."""
+    menu_selection(main_prompt, main_dispatch)
```
Looks good!

charlesrobison commented on students/bcamp3/lesson04/trigrams.py:
```diff
@@ -0,0 +1,101 @@
+    #         print(item)
+    #     else:
+    #         break
+    print(use_trigrams(trigrams))
```
Getting an error when I run this with sherlock_small.txt - what file did you have success with?

charlesrobison commented on students/bcamp3/lesson05/mailroom_pt3.py:
```diff
@@ -0,0 +1,161 @@
+        except KeyboardInterrupt:
+            print("\nPlease use the menu option '4' to Quit.")
+        else:
+            break
```
Works well - good exceptions

bcamp3 commented:
I realize I added features to trigrams.py that limited it to processing
only the full book "sherlock.txt".  It happens in "read_in_data()" where I
look for the beginning and the end of the body of the book.  If those flags
aren't found the data/text isn't read-in.  I've added a check to see if the
first line in the file is the Project Gutenberg book, if it is , process as
usual, if the book header isn't found it processes all lines.  If you would
like I can commit that version.

On Wed, Aug 19, 2020 at 9:49 PM Charles Robison <notifications@github.com>
wrote:

> *@charlesrobison* approved this pull request.
>
> Code is looking and working great - except I can't get trigrams to run.
> What text file are you using? Also, make sure and submit your lesson 5
> files on the course site for grading.
>
> Ready to merge
> ------------------------------
>
> In students/bcamp3/lesson04/mailroom_pt2.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1047#discussion_r473583393>
> :
>
> > +               '   2 - Create a Report.                  \n'
> +               '   3 - Send letters to all donors.       \n'
> +               '   4 - Quit                              \n'
> +               )
> +
> +main_dispatch = {'1': send_thank_you,
> +                 '2': create_report,
> +                 '3': send_letters_to_all_donors,
> +                 '4': quit_msg,
> +                 }
> +
> +# main_dispatch.setdefault('0', 'nothing')
> +
> +if __name__ == "__main__":
> +    """Execute the main menu prompt."""
> +    menu_selection(main_prompt, main_dispatch)
>
> Looks good!
> ------------------------------
>
> In students/bcamp3/lesson04/trigrams.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1047#discussion_r473584917>
> :
>
> > +if __name__ == "__main__":
> +    # get the filename from the command line
> +    try:
> +        filename = sys.argv[1]
> +    except IndexError:
> +        print("You must pass in a filename")
> +        sys.exit(1)
> +    in_data = read_in_data(filename)
> +    words = get_words(in_data)
> +    trigrams = build_trigrams(words)
> +    # for n, item in enumerate(trigrams.items()):
> +    #     if n < 30:
> +    #         print(item)
> +    #     else:
> +    #         break
> +    print(use_trigrams(trigrams))
>
> Getting an error when I run this with sherlock_small.txt - what file did
> you have success with?
> ------------------------------
>
> In students/bcamp3/lesson05/mailroom_pt3.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1047#discussion_r473585591>
> :
>
> > +                 '2': create_report,
> +                 '3': send_letters_to_all_donors,
> +                 '4': quit_msg,
> +                 }
> +
> +# main_dispatch.setdefault('0', 'nothing')
> +
> +if __name__ == "__main__":
> +    """Execute the main menu prompt."""
> +    while True:
> +        try:
> +            menu_selection(main_prompt, main_dispatch)
> +        except KeyboardInterrupt:
> +            print("\nPlease use the menu option '4' to Quit.")
> +        else:
> +            break
>
> Works well - good exceptions
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1047#pullrequestreview-471210532>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHZNACUFIZIK2PP2XPGZ2DSBSTN5ANCNFSM4QAL4VXA>
> .
>

bcamp3 commented on students/bcamp3/lesson04/trigrams.py:
```diff
@@ -0,0 +1,101 @@
+                body = True
+            if line[:7] == '*** END':
+                body = False
+            if body and not line.isupper():
```
At this line the 'body' of the book needs to be identified for data to be read-in.  If 'body' is not found, data is not read.
